src/main/java/edu/wpi/grip/core/SocketHints.java

```diff
@@ -0,0 +1,105 @@
+        }
+
+    }
+
+    public static SocketHint<Mat> createMatSocketHint(final String identifier, final boolean withDefault) {
+        return createObjectSocketHintBuilder(identifier, Mat.class, Mat::new, withDefault).build();
+    }
+
+    public static SocketHint<Size> createSizeSocketHint(final String identifier, final boolean withDefault) {
+        return createObjectSocketHintBuilder(identifier, Size.class, Size::new, withDefault).build();
+    }
+
+
+    public static SocketHint<Point> createPointSocketHint(final String identifier, final boolean withDefault) {
+        return createObjectSocketHintBuilder(identifier, Point.class, Point::new, withDefault).build();
+    }
+
+    public static SocketHint<Number> createNumberSliderSocketHint(final String identifier, final Number number,
+                                                            final Number[] domain) {
+        return createNumberSocketHintBuilder(identifier, number).domain(domain).view(SocketHint.View.SLIDER).build();
+    }
+
+    public static SocketHint<Number> createNumberSpinnerSocketHint(final String identifier, final Number number,
+                                                            final Number[] domain) {
+        return createNumberSocketHintBuilder(identifier, number).domain(domain).view(SocketHint.View.SPINNER).build();
+    }
+
+    public static SocketHint<Number> createNumberSpinnerSocketHint(final String identifier, final Number number) {
+        return createNumberSocketHintBuilder(identifier, number).view(SocketHint.View.SPINNER).build();
+    }
+
+    public static SocketHint<Number> createNumberSocketHint(final String identifier, final Number number) {
+        return createNumberSocketHintBuilder(identifier, number).build();
+    }
+
+    public static SocketHint<Boolean> createBooleanSocketHint(final String identifier, final boolean initialValue) {
+        return new SocketHint.Builder(Boolean.class)
+                .identifier(identifier)
+                .initialValue(initialValue)
+                .view(SocketHint.View.CHECKBOX)
+                .build();
+    }
+
+    public static SocketHint<List> createNumberListRangeSockeHint(final String identifier, final Number low, final Number high) {
+        return new SocketHint.Builder(List.class).identifier(identifier)
+                .initialValueSupplier(() -> Arrays.asList(low, high))
+                .domain(new List[]{Arrays.asList(low, high)})
+                .view(SocketHint.View.RANGE)
+                .build();
+    }
+
+    private static <T> SocketHint.Builder<T> createObjectSocketHintBuilder(final String identifier,
+                                                                   final Class<T> type,
+                                                                   final Supplier<T> supplier,
+                                                                   final boolean withDefault) {
```

Comment:
Instead of having `withDefault`, would it work to just make the supplier an optional parameter, and assume there's no default if it's not specified?